students/duanez2021/lesson04/mailroom_rev02.py

```diff
@@ -0,0 +1,170 @@
+#######################################################################################################
+# 20200625      djm mail room part 1
+# 20200707      djm mail room part 2 (Dictionary implementations)
+# Duane McCollum Python self-paced winter 2020
+#
+# Update your mailroom program to:
+#
+#         Use dicts where appropriate.
+#         See if you can use a dict to switch between the user’s selections.
+#         See if you can use a dict to switch between the users selections.
+#         see Using a Dictionary to switch for what this means.
+#         Convert your main donor data structure to be a dict.
+#         Try to use a dict and the .format() method to produce the letter as one big template,
+#         rather than building up a big string that produces the letter in parts.
+#
+#
+#######################################################################################################
+# original donor db, a list
+import sys, os  # imports go at the top of the file
+donor_db = [("William Gates, III", [653772.32, 12.17]),
+            ("Jeff Bezos", [877.33]),
+            ("Paul Allen", [663.23, 43.87, 1.32]),
+            ("Mark Zuckerberg", [1663.23, 4300.87, 10432.0])
+            ]
```

Comment:
your data structure in lesson04 should be using a dictionary for donors data structure:
```
donor_db = {"Jeff Bezos": [100, 200]}
```